util/comparator.cc

```diff
@@ -98,6 +98,22 @@ class BytewiseComparatorImpl : public Comparator {
     }
     // *key is a run of 0xffs.  Leave it alone.
   }
+
+  virtual bool IsSameLengthImmediateSuccessor(std::string s,
+                                              std::string t) const override {
+    if (s.size() != t.size()) return false;
+    int len = static_cast<int>(s.size());
+    for (int i = len - 1; i >= 0; --i) {
+      const uint8_t byte = s[i];
+      if (byte != static_cast<uint8_t>(0xff)) {
+        s[i] = byte + 1;
```

Comment:
As commented somewhere else, we can't afford to make two string copies here, but should use Slice. So here we need to detect it without changing the string.